terraform/variables.tf

```diff
@@ -173,3 +179,27 @@ variable "distributed_tff_example_worker_2_address" {
   description = "Address of the second worker of the distributed TensorFlow Federated example. Set this when the worker is outside the coordinator mesh."
   type        = string
 }
+
+variable "cross-device" {
+  description = "Enable cross device infrastructure deployment"
+  type        = bool
+  default     = false
+}
+
+variable "spanner_instance_config" {
+  description = "Multi region config value for the Spanner Instance of the cross-device Federated Learning example. Example: 'nam10' for North America."
+  type        = string
+  default     = "regional-europe-west1"
+}
+
+variable "spanner_processing_units" {
+  description = "Spanner's compute capacity of the cross-device Federated Learning example. 1000 processing units = 1 node and must be set as a multiple of 100."
+  type        = number
+  default     = 1000
+}
+
+variable "spanner_database_deletion_protection" {
+  description = "Prevents destruction of the Spanner database of the cross-device Federated Learning example."
+  type        = bool
+  default     = true
```

Comment:
What happens on `terraform destroy`? Why do we default to `true`?

Reply:
Spanner will contain the history of every trained models, so we need to keep the data in Spanner. That's why it's default to true

Reply:
Ok, we need a spanner DB. But why do we want to enable deletion protection by default?

Reply:
Just to keep the history of trained models. But I can default it to false

Reply:
When you set that flag to `true`, it will prevent the deletion of the Spanner database itself, not the data inside the database. I would delete this variable entirely, unless we have a very specific need for exposing it.

Reply:
Done